docs(node pools): updates to node pools content to reflect move to beta (#9460)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/con-config-node-pools.adoc

For a deeper understanding of the node pool configuration options, refer to the link:{BookURLConfiguring}[Strimzi Custom Resource API Reference^].

> NOTE: While the `KafkaNodePools` feature gate that enables node pools is in beta phase, replica and storage configuration properties in the `KafkaNodePool` resource must also be present in the `Kafka` resource. The configuration in the `Kafka` resource is ignored when node pools are used. Similarly, ZooKeeper configuration properties must also be present in the `Kafka` resource when using KRaft mode. These properties are also ignored.

Reviewer: I think this will likely change before node pools leave beta. So maybe we can reword it something like this?

```suggestion
NOTE: Currently, replica and storage configuration properties in the `KafkaNodePool` resource must also be present in the `Kafka` resource. The configuration in the `Kafka` resource is ignored when node pools are used. Similarly, ZooKeeper configuration properties must also be present in the `Kafka` resource when using KRaft mode. These properties are also ignored.
```

.Example configuration for a node pool in a cluster using KRaft mode 
[source,yaml,subs="+attributes"]
----
apiVersion: {KafkaNodePoolApiVersion}
kind: KafkaNodePool
metadata:
  name: kraft-dual-role
  labels:
    strimzi.io/cluster: my-cluster
spec:
  replicas: 3
  roles: # <1>
    - controller
    - broker
  storage:
    type: jbod
    volumes:
      - id: 0
        type: persistent-claim

>         size: 20Gi

Reviewer: Why change the volume size? I do not see any value and 20Gi is in general quite small. I also wonder if we should keep the callouts for the other parts.

Author: That's peculiar. I just changed the order of the examples so that KRaft came first.  I didn't make any changes to the examples themselves. I'll revert this.

Reviewer: Ahh, I did not realize that. But especially for the callouts, it makes sense to have them in the first example rather than the second. It looks good now.
